crates/provider/src/provider/call.rs

```diff
@@ -124,34 +141,79 @@ where
 /// [`Provider::call`]: crate::Provider::call
 #[must_use = "EthCall must be awaited to execute the call"]
 #[derive(Debug, Clone)]
-pub struct EthCall<'req, 'state, T, N>
+pub struct EthCall<'req, 'state, T, N, Resp, Output = Resp, Map = fn(Resp) -> Output>
 where
     T: Transport + Clone,
     N: Network,
+    Resp: RpcReturn,
+    Map: Fn(Resp) -> Output,
 {
     client: WeakClient<T>,
 
     data: &'req N::TransactionRequest,
     overrides: Option<&'state StateOverride>,
     block: Option<BlockId>,
+    method: &'static str,
+    map: Map,
+    _pd: PhantomData<fn() -> (Resp, Output)>,
 }
 
-impl<'req, T, N> EthCall<'req, 'static, T, N>
+impl<'req, T, N, Resp> EthCall<'req, 'static, T, N, Resp>
 where
     T: Transport + Clone,
     N: Network,
+    Resp: RpcReturn,
 {
     /// Create a new CallBuilder.
     pub const fn new(client: WeakClient<T>, data: &'req N::TransactionRequest) -> Self {
-        Self { client, data, overrides: None, block: None }
+        Self {
+            client,
+            data,
+            overrides: None,
+            block: None,
+            method: "eth_call",
+            map: std::convert::identity,
+            _pd: PhantomData,
+        }
+    }
+
+    /// Create new EthCall for gas estimates.
+    pub const fn gas_estimate(client: WeakClient<T>, data: &'req N::TransactionRequest) -> Self {
+        Self {
+            client,
+            data,
+            overrides: None,
+            block: None,
+            method: "eth_estimateGas",
+            map: std::convert::identity,
+            _pd: PhantomData,
+        }
     }
 }
 
-impl<'req, 'state, T, N> EthCall<'req, 'state, T, N>
+impl<'req, 'state, T, N, Resp, Output, Map> EthCall<'req, 'state, T, N, Resp, Output, Map>
 where
     T: Transport + Clone,
     N: Network,
+    Resp: RpcReturn,
+    Map: Fn(Resp) -> Output,
 {
+    /// Map the response to a different type.
+    pub fn map_resp<NewOutput, NewMap: Fn(Resp) -> NewOutput>(
```

Comment:
nit:
idiom elsewhere is to use a where block. i feel like it's easier to read for complex bounds

Reply:
fixed!